.github/workflows/build.yml

```diff
@@ -18,6 +23,9 @@ jobs:
         toolchain: [stable, nightly]
     steps:
       - uses: actions/checkout@main
+      - uses: Swatinem/rust-cache@v2
+        with:
+          save-if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/next' }}
```

Comment:
Could be good to add a quick comment explaining what this does.

Reply:
This actually made me go look into the cache effectiveness. 

I think caching is quite ineffective for many of our jobs because we used nightly for almost all of them. And caching doesn't work on a moving target.

I don't think there is much advantage to using nightly for these jobs, except `rustfmt` as it specifically has extra options available. `clippy` and `check` should probably be using stable?

Reply:
So, the solution is then to switch everything except for `rustfmt` tasks to `stable` - right?

Reply:
Hmm. Okay it is a bit more awkward than that. `make lint` runs `fmt`, `clippy` and `cargo fix`. Which means it now combines stable and nightly builds. But that's probably okay since fmt is "cheap".